src/coreclr/jit/lclvars.cpp

```diff
@@ -3467,7 +3473,7 @@ void Compiler::lvaSortByRefCount()
             assert(varDsc->lvType != TYP_STRUCT ||
                    varDsc->lvDoNotEnregister); // For structs, should have set this when we set lvAddrExposed.
         }
-        else if (varTypeIsStruct(varDsc))
+        if (varTypeIsStruct(varDsc))
```

Comment:
the idea of `if/else` was that each block was setting `lvaSetVarDoNotEnregister` and there was no need to check the rest, now we have blocks that don't set it, so we need to check each.